### DISCUSSION TITLE
Install on CentOS 7 + VirtualMin (Apache httpd reverse proxy + user mode installation)

### DISCUSSION THREAD

**Discussion by ATSandrearicci** [globaleaks/globaleaks-whistleblowing-software#3789] [Q&A] [answered]
It's possible to use GlobalLeaks behind apache configured as reverse proxy.

With Virtualmin create a virtual host with SSL enabled
In Server Configuration -> Website Options set Yes to option "Redirect all requests to SSL site"
In Server Configuration -> Edit Proxy Website set Yes to "Proxying enabled" and "proxy to URL" to https://[FQNServerName]:8443
Optionally you can disable php wich is not required in this virtual server
Server Configuration -> PHP options 

Add manully this directive to enable proxy to https and disable certificate validation in Services -> Configure SSL website
```
SSLProxyEngine on
SSLProxyVerify none
SSLProxyCheckPeerCN off
SSLProxyCheckPeerName off
```
Restart httpd after this changes

python3.6 is defualt installed on CentOS 7.
Install npm as root:
`dnf install npm`
Execute as root:
`npm install -g grunt`

Enter with the linux user owner of the virtual host and download the latest release of Globaleaks with something like:
`curl -L -O https://github.com/globaleaks/GlobaLeaks/archive/refs/tags/v4.13.18.zip`

Unzip the archive and you will get the folder GlobaLeaks-4.13.18
Create a symbolic link (usefull for updates)
`ln -s GlobaLeaks-4.13.18 GlobaLeaks`

Enter in GlobaLeaks/client and build the client:
```
npm install
grunt build
```

Enter in GlobaLeaks/backend and install the bullseye requirements in venv
```
python3 -mvenv env
source env/bin/activate
pip3 install -r requirements/requirements-bullseye.txt
```

Now you can check if your installation is working by doing:
`./bin/globaleaks -n -w ~/GlobaLeaksData/`

If everything is fine it's possible to create a systemd service file like with a command like this in ExecStart
`~/GlobaLeaks/backend/env/bin/python ~/GlobaLeaks/backend/bin/globaleaks -w ~/GlobaLeaksData/`
Indicating in User and Group the unix user owner of the virtual server

You can follow the above steps with RHEL8 and RHEL9 which are more close to debian bookworn.
Just do `dnf module install nodejs:18` isntead of `dnf install npm` and install python requirements bookworm instead of bullseye one.

**Comment by LyraHarring** [comment]
To install an [SSL certificate on CentOS 7](https://certera.com/kb/how-to-install-ssl-certificate-on-centos/) with Virtualmin (Apache reverse proxy in user mode): Go to Virtualmin > Server Configuration > SSL Certificate. You can either upload your certificate and private key manually or use the Let's Encrypt tab to request a free certificate automatically. Virtualmin will handle Apache config and reload it after setup. There are some steps for the installation, which are as follows:- so you need to understand them in detail 

1) Downloading certificates
2) Copying the SSL files
3) Go to the Apache configuration & Find and Edit The ssl.conf or httpd.conf file
4) Restart the Server

I think you need to understand the steps properly in detail, which will guide you through it and help in solving it. Hope it helps!

**Comment by evilaliv3** [comment, marked as answer]
Thank you for the question @ATSandrearicci 

Actually we higly discourage running globaleaks behind a webproxy that terminates TLS as it would invalidate the threat model.

Please let us know if you have any doubt with this regard.
